Source/Dafny/Verifier/Translator.TrStatement.cs

```diff
@@ -505,6 +505,12 @@ public partial class Translator {
         CheckWellformedWithResult(rhs, new WFOptions(null, false, false), rIe, pat.Expr.Type, locals, builder, etran);
         CheckCasePatternShape(pat, rIe, rhs.tok, pat.Expr.Type, builder);
         builder.Add(TrAssumeCmd(pat.tok, Bpl.Expr.Eq(etran.TrExpr(pat.Expr), rIe)));
+      } else if (stmt is HaltRecoveryStatement haltRecoveryStatement) { 
```

Comment:
Would it be possible to add a noVerify annotation to the generated main method, and let this block `throw new NotSupportedException()` ?

Reply:
Good call, I like that better (and there's even precedent for doing that in a rewriter: https://github.com/dafny-lang/dafny/blob/master/Source/Dafny/Rewriter.cs#L1087)